concordium-sdk/src/main/java/com/concordium/sdk/cis2/events/Cis2Event.java

```diff
@@ -0,0 +1,33 @@
+package com.concordium.sdk.cis2.events;
+
+/**
+ * A common API for an event emitted by a CIS2 compliant contract.
+ */
+public interface Cis2Event {
+    /**
+     * Get the type of the CIS2 event.
+     */
+    Type getType();
+
+
+    /**
+     * Types of events that are supported by the CIS2 specification
+     */
+    enum Type {
+        TRANSFER,
+        MINT,
+        BURN,
+        UPDATE_OPERATOR_OF,
+        TOKEN_METADATA,
+        CUSTOM;
+
+        public static Type parse(byte tag) {
+            if (tag == -1) return TRANSFER;
+            if (tag == -2) return MINT;
+            if (tag == -3) return BURN;
+            if (tag == -4) return UPDATE_OPERATOR_OF;
+            if (tag == -5) return TOKEN_METADATA;
+            return CUSTOM;
+        }
+    }
```

Comment:
What is this method? These are not the tags of events we define in the CIS2 standard. The tags there are unsigned integers and while -1 :: i8 if interpreted as a u8 bit pattern is 255, this is confusing for users.

Reply:
I don't understand what you mean here. Java only has signed bytes. 
This is a function for parsing the result of a transaction event into a proper `enum` and hence is not something that is intended for the end user.